File: array/Buy and sell/Buy and sell.go

func maxProfit(prices []int) int {
    var maxProfit,bestBuy = 0, prices[0]
    for i:=0;i<len(prices);i++{
        if prices[i]<bestBuy{
           bestBuy = prices[i]
        }
        profit := prices[i]-bestBuy
        if profit>maxProfit{
            maxProfit = profit
        }
    }
    return maxProfit
}

File: array/Buy and sell/BuyAndSell.js
let price = [7,1,5,3,6,4];

let bestBuy = price[0];
let maxProfit = 0;

for(let i=0;i<price.length;i++){
 if(bestBuy>price[i]){
  bestBuy = price[i]
 }
 profit = price[i]-bestBuy
 console.log(profit)
 if(profit>maxProfit){
  maxProfit = profit
 }
}
console.log(maxProfit)
